### PR TITLE
coretex refactor

### PR DESCRIPTION
**Pull request by tobowers**
just a small refactor to use a switch statement instead of a nested if/else and to allow alternative language program executors to be used.

### CODE REVIEW

File: core/src/cortexStep.ts [hunk 1]
```diff
@@ -65,14 +65,24 @@ function toCamelCase(str: string) {
...
   private readonly memories: WorkingMemory;
   private readonly extraNextActions: NextActions;
+  private readonly processor: LanguageModelProgramExecutor;
+  private readonly options: CoretexStepOptions;
 
-  constructor(entityName: string, pastCortexStep?: PastCortexStep) {
+  constructor(
+    entityName: string,
+    pastCortexStep?: PastCortexStep,
+    options?: CoretexStepOptions
```
Comment (kafischer) [inline core/src/cortexStep.ts:84]: this doesn't work right to have two optionals in a row like that?

**Review by kafischer — CHANGES_REQUESTED** [review]

File: core/src/cortexStep.ts (hunk 1 above)
Comment (tobowers) [reply core/src/cortexStep.ts:84]: it does work, but it's a little awkward syntax maybe for the first one.

Comment (tobowers) [reply core/src/cortexStep.ts:84]: the tests pass btw.

Comment (kafischer) [reply core/src/cortexStep.ts:84]: so like you have to pass `new CortexStep('bogus', {[]}, options)`

File: core/src/cortexStep.ts [hunk 2]
```diff
@@ -66,49 +62,43 @@ function toCamelCase(str: string) {
 }
 
 interface CoretexStepOptions {
+  pastCoretexStep?: CortexStep;
```
Comment (kafischer) [inline core/src/cortexStep.ts:65]: minor typo pastCortexStep and CoretexStepOptions

Comment (tobowers) [reply core/src/cortexStep.ts]: fixorated

File: core/src/cortexStep.ts (hunk 1 above)
Comment (tobowers) [reply core/src/cortexStep.ts]: resolving this as API has changed.

**Review by kafischer — APPROVED** [review]
sweet!

**Pull request merged**